k8s/upgrades/0.5.x-0.6.0/controller.patch.tpl.yml

```diff
@@ -0,0 +1,71 @@
+{
+  "metadata": {
+     "labels": {
+       "openebs/volume-provisioner": "jiva",
+       "pvce": "@pvc-name"
+     }
+  },
+  "spec": {
+     "selector": {
+        "matchLabels": {
+           "openebs/volume-provisioner": "jiva",
+           "pvc": "@pvc-name"
+        }
+     },
+     "template": {
+        "metadata": {
+           "labels": {
+              "openebs/volume-provisioner": "jiva",
+              "pvc": "@pvc-name"
+           }
+        },
+        "spec": {
+           "containers":[
+              {
+                "name": "@c_name",
+                "image": "openebs/jiva:0.6.0",
+                "env":[
+                  {
+                    "name": "REPLICATION_FACTOR",
+                    "value": "@rep_count"
+                  }
+                ]
+              },
+              {
+                "name": "maya-volume-exporter",
+                "command": [
+                   "maya-exporter"
+                ],
+                "image": "openebs/m-exporter:0.6.0"
+              }
+           ], 
+          "tolerations": [
+            {
```

Comment:
Do you think we can make controller sticky for below taints cases:
- disk pressure
- memory pressure
- out of disk

Reasoning being, `the less the controller jitters the more available is the storage`. Let other pods get evicted due to these taints. This plays the trick of masquerading the controller into getting a higher PriorityClass than non-openebs Pods.

If we invert this logic, and assume other operators & pods are making use of high PriorityClass or able to ward off evictions via above mentioned trick, & openebs controller does not make use of this trick, then openebs controller will be the first to be evicted when these problems arise.

Reply:
Just thinking about this @AmitKumarDas .. I guess the reasons with adding these for replicas is: "even if we are not able to serve writes under the conditions, still don't move in order to maintain relation w/ hostPath, we will eventually sync".  With controller, it becomes important to ensure target can always take I/O requests. By retaining the controller on the node by toleration, the system issues (memory pressure, unreachable etc.,) may render it dysfunctional. In which case, it is desirable to have it reschedule as early as possible (your latter comment about getting this the 1st one to be evicted may then be more helpful)

Reply:
Well.. I guess Kubernetes will always want to evict and if Jiva controller is always moving around it does not have a chance to serve IO. so let it move fast for certain taints. Let it ignore some other taints and so on.  Alternatively if we think it should move fast then let it survival to toleration as 0 seconds to move real fast.. as always tests and various scenarios will help us get the right combinations...

Reply:
This patch is mainly focusing on providing the taints that were introduced on target/controller and replica in 0.6 provisioned PVs. Should we still go ahead and add additional taints @AmitKumarDas which are not applicable to 0.6 PVs? Or probably first have these taints introduced in 0.7 and let upgrade to 0.7 take care of it?